components/ui/Icon.tsx

```diff
@@ -127,208 +35,83 @@ export function useIconProps(props: Partial<TablerIconProps>): TablerIconProps {
 	return useProps("Icon", defaultStyleProps, props);
 }
 
-const Icon = forwardRef<_TablerIconType, IconProps>(function Icon(
-	{type, size, stroke, color, ...rest}: IconProps,
+// Helper function to convert icon name to proper import name
+const getIconImportName = (iconName: string): string => {
+	// Handle special cases first
+	// These are needed to have full backwards compatibility
+	const specialCases: Record<string, string> = {
+		"icon-hand-finger": "IconHandFinger",
+		"icon-info-circle-filled": "IconInfoCircleFilled",
+		"paper-clip": "IconPaperclip", // lowercase 'c'
+	};
+
+	if (specialCases[iconName]) {
+		return specialCases[iconName];
+	}
+	// Convert kebab-case or other formats to PascalCase with Icon prefix
+	const pascalCase = iconName
+		.split(/[-_\s]/)
+		.map(
+			(word) =>
+				word.charAt(0).toUpperCase() + word.slice(1).toLowerCase(),
+		)
+		.join("");
+
+	return `Icon${pascalCase}`;
+};
+
+const Icon = forwardRef<TablerIcon, IconProps>(function Icon(
+	{iconType, size, stroke, color, ...rest}: IconProps,
 	ref,
 ) {
 	const theme = useMantineTheme();
+	const [IconComponent, setIconComponent] =
+		useState<React.ComponentType<TablerIconProps> | null>(null);
+
+	useEffect(() => {
+		if (typeof iconType === "string") {
+			const iconImportName = getIconImportName(iconType);
+
+			// Dynamic import of the icon
+			import("@tabler/icons-react")
+				.then((module) => {
```

Comment:
@MajorMeerkatThe3rd a question about this: Will this load all icons into the web application at once? If so, could you review how it influences the bundle size?

Reply:
@MajorMeerkatThe3rd wow, very interesting, just learned something new, many thanks!!!